out/blocks/wd_admin_payment_main_form.tpl

```diff
@@ -57,71 +57,86 @@
-    <tr>
+  <tr>
+    <td><img src="[{$logoUrl}]" alt="[{$edit->oxpayments__oxdesc->value}]"></td>
+  </tr>
   [{/if}]
 
   <tr>
     <td height="40" valign="top">[{ $edit->oxpayments__oxdesc->value }]</td>
-  <tr>
-[{/if}]
+  </tr>
+  [{/if}]
 
 [{$smarty.block.parent}]
 
 [{if $configFields}]
   [{foreach from=$configFields key=configKey item=configField}]
-    [{assign var="fieldName" value=$configField.field}]
-    <tr>
-      <td class="edittext" width="70">[{$configField.title}]</td>
-      <td class="edittext">
-        [{if $configField.type === 'text'}]
-          <input id="[{$configKey}]" type="text" class="editinput" size="25" name="editval[[{$fieldName}]]" value="[{$edit->$fieldName->value}]">
-        [{/if}]
-
-        [{if $configField.type === 'select'}]
-          <select name="editval[[{$fieldName}]]">
-            [{foreach from=$configField.options key=optionKey item=optionValue}]
-              <option value="[{$optionKey}]" [{if $edit->$fieldName->value == $optionKey}]selected[{/if}]>[{$optionValue}]</option>
-            [{/foreach}]
-          </select>
-        [{/if}]
-
-        [{if $configField.description}]
-          [{$oViewConf->getInputHelpHtml($configField.description)}]
-        [{/if}]
-      </td>
-    </tr>
-
-    [{if $configKey === 'httpPassword' && $configFields.apiUrl && $configFields.httpUser}]
-      <tr>
-        <td class="edittext" width="100">
-          <input type="button" value="[{oxmultilang ident="test_credentials"}]" onclick="wdTestPaymentMethodCredentials()">
-        </td>
-        <td>
-          <span id="test_credentials_result"></span>
-        </td>
-      </tr>
-    [{/if}]
+  [{assign var="fieldName" value=$configField.field}]
+  <tr>
+    <td class="edittext" width="70">[{$configField.title}]</td>
+    <td class="edittext">
+      [{if $configField.type === 'text'}]
+    <input id="[{$configKey}]" type="text" class="editinput" size="25" name="editval[[{$fieldName}]]"
+           value="[{$edit->$fieldName->value}]">
+      [{/if}]
+
+      [{if $configField.type === 'select'}]
+      <select name="editval[[{$fieldName}]]">
+        [{foreach from=$configField.options key=optionKey item=optionValue}]
+        <option value="[{$optionKey}]" [{if $edit->$fieldName->value == $optionKey}]selected[{/if}]>[{$optionValue}]
+        </option>
+        [{/foreach}]
+      </select>
+      [{/if}]
+
+      [{if $configField.description}]
+      [{$oViewConf->getInputHelpHtml($configField.description)}]
+      [{/if}]
+    </td>
+  </tr>
+
+  [{if $configKey === 'httpPassword' && $configFields.apiUrl && $configFields.httpUser}]
+  <tr>
+    <td class="edittext" width="100">
+      <input type="button" value="[{oxmultilang ident="test_credentials"}]" onclick="wdTestPaymentMethodCredentials()">
+    </td>
+    <td>
+      <span id="test_credentials_result"></span>
+    </td>
+  </tr>
+  [{/if}]
   [{/foreach}]
-[{/if}]
+  [{if $edit->oxpayments__oxid->value === 'wdcreditcard'}]
```

Comment:
I think it would be cleaner if this was in `Payment_Method::getConfigFields`. You could create a new field type for this, e.g. "info".

Reply:
:+1: